# Тести для малих вибірок та пропорцій



**Завдання 1**. E-commerce компанія після редизайну сайту підозрює, що **середній час до покупки (time-to-purchase)** користувачів **збільшився**.
Історично середній час від першого заходу на сайт до покупки становив $\mu_0 = 123.7$ хвилин.

Команда аналітиків випадково обрала дані **7 покупців після редизайну**:
`128, 135, 121, 142, 126, 151, 123`

З допомогою наявних даних зʼясуйте, чи збільшився середній час покупки після редизайну.

Для цього виконайте наступні 6 кроків. Правильне виконання кожного з кроків оцінюється в 1 бал.

1. Запишіть нульову та альтернативну гіпотези і визначте тип тесту.
2. Обчисліть вибіркові статистики: $\bar x$, $s$, $n$.
3. Оберіть тип тесту та виконайте його (знайдіть критичне значення тесту, статистику тесту та р-значення) будь-яким пасуючим способом, щоб перевірити гіпотезу на рівні значущості $\alpha = 0.10$.
4. Прийміть рішення, чи відхиляєте ви гіпотезу $H_0$ використовуючи p-value.
5. Напишіть висновок: чи справді редизайн сайту подовжив час до покупки?
6. Чи зміниться ваше рішення при зміні рівня значущості на $\alpha = 0.05$.


1. $H_0$: $\mu$ = 123.7
$H_a$: $\mu$ > 123.7
Це правосторонній тест

In [ ]:
#2
import numpy as np
from scipy import stats
avg_time = np.array([128, 135, 121, 142, 126, 151, 123])
mu0 = 123.7
x_bar = np.mean(avg_time)
s = np.std(avg_time, ddof=1) 
n = len(avg_time)

In [17]:
#3
alpha = 0.10
t_stat = (x_bar - mu0) / (s / np.sqrt(n))
df = n - 1
t_crit = stats.t.ppf(alpha, df)
p_value = stats.t.cdf(t_stat, df)

In [18]:
#4
res = stats.ttest_1samp(avg_time, popmean=mu0, alternative="greater")
print(f"scipy: t = {res.statistic:.3f}, p-value = {res.pvalue:.5f}")

if res.pvalue < alpha:
    print(f"Відхиляємо H0: середній час до покупки збільшився на рівні значущості {alpha}")
else:
    print("Не відхиляємо H0: доказів недостатньо")

scipy: t = 2.069, p-value = 0.04202
Відхиляємо H0: середній час до покупки збільшився на рівні значущості 0.1


5. Редизайн сайту справді подовжив час до покупки.

6. При зміні рівня значущості на 0.05 рішення залишається таким самим.

**Завдання 2.**

До спеціальної рекламної кампанії **23%** дорослих упізнавали логотип компанії. Після завершення кампанії відділ маркетингу провів опитування: з **1200** випадково відібраних дорослих **311** упізнали логотип.

Перевірте на рівні значущості **3%** ($\alpha=0.03$), чи дають ці дані достатні підстави стверджувати, що **тепер більше ніж 23%** дорослих упізнають логотип компанії. Для розвʼязку використовуйте бібліотеку `statsmodels`.

Зробіть висновок, чи зросла впізнаваність логотипу.

Додатково, обчисліть довірчий інтревал на заданому рівні значущості і проінтерпретуйте текстом - як він додатково пояснює прийняте нами рішення?

In [2]:
import statsmodels.api as sm
from statsmodels.stats.proportion import proportions_ztest, proportion_confint

count = 311   # кількість "успіхів"
nobs = 1200    # обсяг вибірки
p0 = 0.23      # гіпотетична пропорція (H0)

# z-тест для пропорції
stat, pval = proportions_ztest(count, nobs, p0, alternative='larger')
confint = proportion_confint(count, nobs, alpha=0.03, method='normal')

print("🔹 Тест для однієї пропорції")
print(f"Z-статистика = {stat:.3f}")
print(f"p-value = {pval:.4f}")
print(f"97% довірчий інтервал: {confint}")

🔹 Тест для однієї пропорції
Z-статистика = 2.306
p-value = 0.0106
97% довірчий інтервал: (0.23171700302179205, 0.28661633031154127)


Інтерпретуємо результати:

1. Односторонній тест
    H0 : p = 0.23
    H1 : p > 0.23
    Оскільки p-value = 0.0106 < 0.03, відхиляємо H0. Є статистично значущі докази, що частка тих, хто впізнає логотип, більша за 23%.
2. 97% довірчий інтервал (0.231, 0.286)
    Це означає, що з імовірністю 97% справжня частка лежить у межах від 23.1% до 28.6%.
    Нижня межа майже на 0.23, тому в двосторонньому тесті ми б не відхилили H0.
    Але в односторонньому тесті — ми вже отримуємо значущість.